agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java

```diff
@@ -935,7 +935,7 @@ public void validate(ProcessorType processorType, IncludeExclude includeExclude)
                   + " processors here: https://go.microsoft.com/fwlink/?linkid=2151557");
         }
         if (matchType == MatchType.REGEXP && attribute.value != null) {
-          validateRegex(attribute.value, processorType);
+          validateRegex(String.valueOf(attribute.value), processorType);
```

Comment:
we should probably use same format we use in the exporter:

https://github.com/Azure/azure-sdk-for-java/blob/0f379118d823884583bd43f1aa4f5e6ac9decab9/sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/Mappings.java#L58

Reply:
https://github.com/Azure/azure-sdk-for-java/pull/37712

Reply:
probably no. that will be super complicated: regex on nonstring.